test/object-store/util/test_file.hpp

```diff
@@ -347,6 +348,11 @@ class TestAppSession {
         return m_app->sync_manager();
     }
 
+    // Close the app instance (or tear down the TestAppSession)
+    void close(bool tear_down = false);
+    // Re-open the app instance using app_config
+    void reopen(bool log_in = false);
```

Comment:
why is this needed? 🤔 Can't you create a new TestAppSession if needed?

Reply:
This is so I can shut down and restart the app and retain the same metadata realm and logged in user in order to simulate the "restart client app with cached user" issue where the location info hasn't been queried yet when the sync session is started.
If I create a new TestAppSession, it will create a new directory and a new logged in user (which will request the location info in the process).

Reply:
got it. it may be worth being able to create a TestAppSession that retains the same metadata realm (i.e, by passing the directory and check if the realm exists, otherwise start from fresh).

Reply:
I would prefer to have TestAppSession remain a strictly scoped type and have a way to reuse a metadata directory between instances, but it's also not a big deal and not worth spending much time on.

Reply:
I can update this in a future PR.